### PR TITLE
Mongo

### PR DESCRIPTION
**Pull request by andrew-d-taylor**
Before this is merged into master and then heroku, we should make sure that our MONGODB_URI environment variable is set correctly.

### CODE REVIEW

**Comment by andrew-d-taylor** [comment]
Pending commit that includes tests

File: splatter.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+class SplatterService(object):
+
+    def __init__(self, database):
+        self.database = database
+
+    def create_api_key(self, name, key=None):
+
+        if key is not None and len(str(key)) != api_key_length:
+            raise ValueError("API key must be %s characters long")
```
Comment (andrew-d-taylor) [inline splatter.py:16]: Missing string substitution

File: app.py [hunk 2]
```diff
@@ -8,11 +8,32 @@
...
+###
+# Configure the database client
+# MONGODB_URI should be of the format mongodb://<dbuser>:<dbpass>@<host>:<port>/<dbname>
+###
+logger = app.logger
+mongodb_uri = os.environ['MONGODB_URI'] if 'MONGODB_URI' in os.environ else 'mongodb://localhost:27017/splatpal'
+logger.info("Attempting initial connection to Mongo at URI: %s", mongodb_uri)
+database = MongoClient(mongodb_uri)
+try:
+    # The official recommended hack to determine if the client connected successfully,
```
Comment (Sidetalker) [inline app.py:28]: love it

File: splatter.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+
+import uuid
+from pymongo import MongoClient
+
+api_key_length = 30
+
+
+class SplatterService(object):
```
Comment (Sidetalker) [inline splatter.py:8]: Let's keep naming schemes for the top level objects/services/whatever to `Splat`thing so `SplatService`

File: splatter.py (hunk 1 above)
Comment (Sidetalker) [reply splatter.py:16]: What do you mean by that? We should skip validation on explicitly included keys so I can have an API key that's the middle finger emoji if I want

File: splatter.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        self.database = database
+
+    def create_api_key(self, name, key=None):
+
+        if key is not None and len(str(key)) != api_key_length:
+            raise ValueError("API key must be %s characters long")
+
+        if key is None:
+            key = uuid.uuid4()
+            key = str(key)[:30]
```
Comment (Sidetalker) [inline splatter.py:20]: You should probably use `api_key_length` here

Comment (Sidetalker) [reply splatter.py:20]: Actually you know what let's just use the standard 38 char 128-bit uuid generated by `uuid.uuid4()` - still allow custom ones if we want but I don't think we have any real excuse for trimming a valid UUID

**Review by Sidetalker — CHANGES_REQUESTED** [review]
Looks good. Complete tests and reopen in a feature branch targeting develop per git-flow http://danielkummer.github.io/git-flow-cheatsheet/

**Pull request closed without merging**